GVRf/IO/STL/src/org/gearvrf/stl/STLController.java

```diff
@@ -0,0 +1,401 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.stl;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import android.content.Context;
+import android.hardware.input.InputManager;
+import android.hardware.input.InputManager.InputDeviceListener;
+import android.util.Log;
+import android.view.InputDevice;
+import android.view.KeyEvent;
+import android.view.MotionEvent;
+
+/**
+ * Use this class to translate MotionEvents and KeyEvents generated by the
+ * controller to values that can be used to create InputEvents for the GVRf
+ * GVRInputManager.
+ */
+public class STLController implements InputDeviceListener {
```

Comment:
I wanted to keep the device specific logic out of the GVRInputManager. The InputManager needs to be generic enough and not turn into an InputDeviceManager.

The plan is to grow the STL instead of the InputManager to support other IO devices, thereby minimizing the changes to the GVRf Framework.